rs/replicated_state/src/canister_state/queues/tests.rs

```diff
@@ -772,6 +907,166 @@ fn test_skip_input_round_robin() {
     assert!(loop_detector.detected_loop(&queues));
 }
 
+#[test]
+fn test_peek_input_with_stale_references() {
```

Comment:
Could you put a comment in the test code, which message gets stale and when? It's hard to follow for someone with no context knowledge